src/dscanner/analysis/objectconst.d

```diff
@@ -5,100 +5,89 @@
 
 module dscanner.analysis.objectconst;
 
-import std.stdio;
-import std.regex;
-import dparse.ast;
-import dparse.lexer;
 import dscanner.analysis.base;
 import dscanner.analysis.helpers;
-import dsymbol.scope_ : Scope;
+import std.stdio;
 
-/**
- * Checks that opEquals, opCmp, toHash, 'opCast', and toString are either const,
- * immutable, or inout.
- */
-final class ObjectConstCheck : BaseAnalyzer
+extern(C++) class ObjectConstCheck(AST) : BaseAnalyzerDmd!AST
 {
-	alias visit = BaseAnalyzer.visit;
-
 	mixin AnalyzerInfo!"object_const_check";
+	alias visit = BaseAnalyzerDmd!AST.visit;
 
-	///
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	extern(D) this(string fileName)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName);
 	}
 
-	mixin visitTemplate!ClassDeclaration;
-	mixin visitTemplate!InterfaceDeclaration;
-	mixin visitTemplate!UnionDeclaration;
-	mixin visitTemplate!StructDeclaration;
-
-	override void visit(const AttributeDeclaration d)
+	void visitAggregate(AST.AggregateDeclaration ad)
 	{
-		if (d.attribute.attribute == tok!"const" && inAggregate)
-		{
-			constColon = true;
-		}
-		d.accept(this);
-	}
+		import dmd.astenums : MODFlags, STC;
 
-	override void visit(const Declaration d)
-	{
-		import std.algorithm : any;
-		bool setConstBlock;
-		if (inAggregate && d.attributes && d.attributes.any!(a => a.attribute == tok!"const"))
-		{
-			setConstBlock = true;
-			constBlock = true;
-		}
-
-		bool containsDisable(A)(const A[] attribs)
-		{
-			import std.algorithm.searching : canFind;
-			return attribs.canFind!(a => a.atAttribute !is null &&
-				a.atAttribute.identifier.text == "disable");
-		}
+		if (!ad.members)
+			return;
 
-		if (const FunctionDeclaration fd = d.functionDeclaration)
+		foreach(member; *ad.members)
 		{
-			const isDeclationDisabled = containsDisable(d.attributes) ||
-				containsDisable(fd.memberFunctionAttributes);
-
-			if (inAggregate && !constColon && !constBlock && !isDeclationDisabled
-					&& isInteresting(fd.name.text) && !hasConst(fd.memberFunctionAttributes))
+			if (auto fd = member.isFuncDeclaration())
+			{
+				if (fd.type && !fd.type.mod == MODFlags.const_ && isInteresting(fd.ident.toString()) && 
+					!(fd.storage_class & STC.disable))
```

Comment:
Why are disabled functions exempt from this check?

Reply:
This is the behavior of DScanner. Probably if you are not going to use a function anyways it does not really matter if you declared it const or not